File: api/src/dto/client.dto.ts
import { ValidateNested } from 'class-validator';
import CredentialsDTO from './credentials.dto';
import PersonDTO from "./person.dto"
import SessionsDTO from './sessions.dto';

export default class ClientDTO {

    public id?: string 

    @ValidateNested()  
    public personDTO?: PersonDTO;

    @ValidateNested()
    public credentialsDTO?: CredentialsDTO;

    public sessionsDTO?: SessionsDTO[];
}

File: mobile/lib/src/features/auth/data/dto/client_dto.dart
import 'package:basearch/src/features/auth/data/dto/credentials_dto.dart';
import 'package:basearch/src/features/auth/data/dto/person_dto.dart';

class ClientDTO {
  const ClientDTO(this.id, this.credentialsDTO, this.personDto);
  final String? id;
  final CredentialsDTO? credentialsDTO;
  final PersonDTO? personDto;
}
